sgt-back-end/index.js

```diff
@@ -0,0 +1,171 @@
+    .then(result => {
+      if (!result) {
+        res.status(500).json({
+          error: 'Cannot find list of grades'
+        });
+      } else {
+        res.status(200).json(result.rows);
+      }
+    });
+});
+
+app.get('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = parseInt(req.params.gradeId, 10);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    res.status(400).json({
+      error: '"gradeId" must be a positive integer'
+    });
+    return;
+  }
+  const sql = `
+  select *
+    from "grades"
+   where "gradeId" = $1
+   `;
+  const params = [gradeId];
+  db.query(sql, params)
+    .then(result => {
+      const grade = result.rows[0];
+      if (!grade) {
+        res.status(404).json({
+          error: `Cannot find grade with "gradeId" ${gradeId}`
+        });
+      } else {
+        res.json(grade);
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occured'
+      });
+    });
+});
+
+app.post('/api/grades', (req, res, next) => {
+  const gradeId = parseInt(req.params.gradeId, 10);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    return res.status(400).json({
+      error: 'invalid gradeId'
+    });
+  } else if (!req.body.name || !req.body.course || !req.body.score) {
+    return res.status(400).json({
+      error: 'Invalid Data Values OR not all data requirements met'
+    });
+  }
+  const sql = `
+  insert into "grades" ("name", "course", "score")
+  values ($1, $2, $3)
+  returning *;
+  `;
+
+  const params = [req.body.name, req.body.course, req.body.score];
+  db.query(sql, params)
+    .then(result => {
+      const newRow = result.rows[0];
+      if (!params) {
+        res.status(404).json({
+          error: 'an Unexpected error has occured'
+        });
+      } else {
+        res.json(newRow);
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An Unexpected error occurred.'
+      });
+    });
+});
+
+app.put('/api/grades/:gradeId', (req, res, next) => {
+  if (!req.params.gradeId || req.params.gradeId <= 0) {
+    return res.status(404).json({
+      error: 'Invalid gradeID OR gradeId does not exist'
+    });
+  } else if (req.body.score <= 0) {
+    return res.status(404).json({
+      error: 'Invalid score, score must be a non-negative number'
+    });
+  } else if (!req.body.course) {
+    return res.status(400).json({
+      error: 'score is missing/does not exist'
+    });
+  }
+  const sql = `
+  update "grades"
+      set "score" = $1
+    where "gradeId" = $2 AND "name" = $3 AND "course" = $4
+    returning *;
+    `;
+  const params = [req.body.score, req.params.gradeId, req.body.name, req.body.course];
+  db.query(sql, params)
+    .then(result => {
+      const updateRow = result.rows[0];
+      if (!updateRow) {
+        return res.status(404).json({
+          error: 'update failed, gradeId and name may not match/exist'
```

Comment:
This error message is not communicating very much. If you found no row to update, then `"may not"` is inaccurate.